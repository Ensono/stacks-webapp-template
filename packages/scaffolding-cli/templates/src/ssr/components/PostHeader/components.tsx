import styled from "@emotion/styled"

export const ImgHolder = styled.div`
    height: 250px;
    border: 2px solid rgba(0, 0, 0, 0.12);
    margin: 8px 8px 8px 0;
    border-radius: 4px;
    margin-bottom: 8px;
    overflow: hidden;
`

export const Image = styled.img`
    display: block;
    margin: auto;
`
