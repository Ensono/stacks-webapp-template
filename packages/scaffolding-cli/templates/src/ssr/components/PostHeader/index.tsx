import Avatar from "../Avatar"
import Date from "../Date"
import {ImgHolder, Image} from "./components"
import React from "react"

export default function PostHeader({title, coverImage, date, author}) {
    return (
        <>
            <br />
            <h1>{title}</h1>
            <div>
                {author && (
                    <Avatar name={author.name} picture={author.picture} />
                )}
                <div>
                    <Date dateString={date} />
                </div>
            </div>
            <ImgHolder>
                <Image title={title} src={coverImage.url} />
            </ImgHolder>
        </>
    )
}
