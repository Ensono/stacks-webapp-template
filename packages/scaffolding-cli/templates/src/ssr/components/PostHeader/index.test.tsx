import {render} from "@testing-library/react"
import React from "react"
import PostHeader from "."

test("renders ApiPane snapshot", () => {
    const {asFragment} = render(
        <PostHeader
            title=""
            coverImage=""
            date="01 Jan 1970 00:00:00 GMT"
            author=""
        />,
    )
    expect(asFragment()).toMatchSnapshot()
})
