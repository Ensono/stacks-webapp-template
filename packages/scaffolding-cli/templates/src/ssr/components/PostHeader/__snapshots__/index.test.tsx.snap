// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ApiPane snapshot 1`] = `
<DocumentFragment>
  <br />
  <h1 />
  <div>
    <div>
      <time
        datetime="01 Jan 1970 00:00:00 GMT"
      >
        01-01-1970
      </time>
    </div>
  </div>
  <div
    class="css-wdz4ep"
  >
    <img
      class="css-1s0bohf"
      title=""
    />
  </div>
</DocumentFragment>
`;
